server/container/util-java8/src/test/java/org/apache/james/util/FunctionGeneratorTest.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.function.Function;
+
+import org.junit.Test;
+
+public class FunctionGeneratorTest {
+
+    @Test
+    public void combineShouldCombine() {
```

Comment:
Ok, I see why it needs to be explained.

I did not succeeded to name this correctly because the concept was not clear to me either. I knew it was needed, but I did not knew why. Now I have better insight.

 - The operation is similar to function composition
 - Reducing Function using compose() directly works well

Now :

 - You might have Function generated dynamically
 - But knows how to compose them whatever the generated functions are
 - As you do not have the functions yet you can not compose them yet
 - But you can chain the generators so that the result is the composition of generated functions.

Naming **FunctionGenerator::composeGeneratedFunctions** better reflect this.